packages/syft/src/syft/service/job/job_stash.py

```diff
@@ -388,8 +388,12 @@ def wait(self):
         if self.resolved:
             return self.resolve
 
+        if subjobs:
+            self.result.wait()
```

Comment:
maybe this is not the best way to do it. What if the subjobs are still running but somehow you are already returning the result?

Reply:
Hmm, yeah. I'm starting to think that subjobs is not the proper name for this flag. Since we're aiming to wait for only the subjobs that we're waiting the results.

Reply:
In mean. We should wait only for those subjobs we are directly dependent on its results, right?

Reply:
oke that makes sense